### PR TITLE
[MNTR] Add include and exclude test filter feature

### LINKED ISSUE #4907: MNTR: make spec filtering more flexible
Version: 1.4.18

The spec filtering we have for the MNTR today is not very well documented and is extremely brittle - i.e. I have to get the FQN of the method marked with `MultiNodeFact` spelled exactly right on the CLI. https://getakka.net/articles/testing/multi-node-testing.html#running-multi-node-tests

I would strongly prefer it if the spec argument scheme followed what NBench does, which allows wildcard include / exclude filtering rules for this purpose: https://nbench.io/articles/running.html#commandline-arguments

### PR DESCRIPTION
**Pull request by Arkatufus**
Closes  #4907

### CODE REVIEW

File: src/core/Akka.MultiNodeTestRunner/Program.cs [hunk 1]
```diff
@@ -130,7 +144,7 @@ static void Main(string[] args)
             var suiteName = Path.GetFileNameWithoutExtension(Path.GetFullPath(args[0].Trim('"')));
             var teamCityFormattingOn = CommandLine.GetPropertyOrDefault("multinode.teamcity", "false");
             if (!Boolean.TryParse(teamCityFormattingOn, out TeamCityFormattingOn))
-                throw new ArgumentException("Invalid argument provided for -Dteamcity");
+                throw new ArgumentException("Invalid argument provided for -Dmultinode.teamcity");
```
Comment (Aaronontheweb) [inline src/core/Akka.MultiNodeTestRunner/Program.cs:147]: nice catch

File: src/core/Akka.MultiNodeTestRunner/README.md [hunk 2]
```diff
@@ -8,13 +8,30 @@ This README explains how to run the `Akka.MultiNodeTestRunner` to execute any `M
...
+- __-Dmultinode.listen-address__ : The TCP/IP address the multi node test runner should listen for test node reports/logs. Default value is `127.0.0.1`.
+- __-Dmultinode.listen-port__ : The TCP/IP port the multi node test runner should listen for test node reports/logs. Default value is `6577`.
+- __-Dmultinode.reporter__ : The report type this runner should export in. Valid values are `trx`, `teamcity`, and `console`. Note that report files are exported to the current directory for `trx`. Default value is `console`.
+- __-Dmultinode.clear-output__ : This flag will clear the output folder before any test is run when it is set to 1. Default value is 0.
+- __-Dmultinode.spec__ : Apply a filter to the test class names within the dll. Any fully qualified test class name that contains this string will run. Default value is (all).
+- __-Dmultinode.include__ : a "`,`" (comma) separted list of wildcard pattern to be matched and included in the tests. Default value is `*` (all). The filter is applied on the complete name of the test `Namespace.Class+MethodName`
```
Comment (Aaronontheweb) [inline src/core/Akka.MultiNodeTestRunner/README.md]: Are you sure it's not just the `MethodName`?

**Review by Aaronontheweb — COMMENTED** [review]
Left a comment

File: src/core/Akka.MultiNodeTestRunner/Program.cs [hunk 3]
```diff
@@ -221,25 +238,41 @@ static void Main(string[] args)
...
+                                CultureInfo.InvariantCulture.CompareInfo.IndexOf(node.TestName,
                                     specName,
                                     CompareOptions.IgnoreCase) < 0)
                             {
-                                PublishRunnerMessage($"Skipping [{test.Value.First().MethodName}] (Filtering)");
+                                PublishRunnerMessage($"Skipping [{node.MethodName}] (Filtering)");
+                                continue;
+                            }
+
+                            var fullName = $"{node.TypeName}+{node.MethodName}";
```
Comment (Arkatufus) [inline src/core/Akka.MultiNodeTestRunner/Program.cs:257]: Full name of the test

File: src/core/Akka.MultiNodeTestRunner/Program.cs [hunk 4]
```diff
@@ -221,25 +238,41 @@ static void Main(string[] args)
...
+
+                            // include filter
+                            if (!include.IsMatch(fullName))
```
Comment (Arkatufus) [inline src/core/Akka.MultiNodeTestRunner/Program.cs:260]: Run the set of regexes on the full name.

File: src/core/Akka.MultiNodeTestRunner/Program.cs [hunk 5]
```diff
@@ -221,25 +238,41 @@ static void Main(string[] args)
...
+                            {
+                                PublishRunnerMessage($"Skipping [{fullName}] (Include filter)");
+                                continue;
+                            }
+
+                            if (exclude.IsMatch(fullName))
```
Comment (Arkatufus) [inline src/core/Akka.MultiNodeTestRunner/Program.cs:266]: Run the set of regexes on the full name.

File: src/core/Akka.MultiNodeTestRunner/README.md (hunk 2 above)
Comment (Arkatufus) [reply src/core/Akka.MultiNodeTestRunner/README.md]: Yes, I've made sure of that.

File: src/core/Akka.MultiNodeTestRunner/Program.cs (hunk 3 above)
Comment (Aaronontheweb) [reply src/core/Akka.MultiNodeTestRunner/Program.cs:257]: Why not just do one or the other? Each test class usually only has a single multinodefact due to XUnit's insistence on it

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**